### PR TITLE
restic-browser: init at 0.3.1

### PR DESCRIPTION
**Pull request by js6pak**
https://github.com/emuell/restic-browser

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by KiaraGrouwstra** [comment]
Result of `nixpkgs-review pr 348387` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>restic-browser</li>
  </ul>
</details>

**Comment by KiaraGrouwstra** [comment]
the package builds for me, tho on run-time i still get:

```
ac: Unknown GPU, using 0 for raster_config
Cannot find target for triple amdgcn-- Unable to find target for this triple (no targets are registered)
[1]    1672486 segmentation fault (core dumped)  results/restic-browser/bin/restic-browser --help
```

**Comment by KiaraGrouwstra** [comment]
hm, still getting the same thing on run-time, so i wonder if that's on me.

fwiw, `lshw` describes my GPU as follows (as the error mentioned that):

```
           *-display
                product: amdgpudrmfb
                physical id: 0
                bus info: pci@0000:04:00.0
                logical name: /dev/fb0
                version: c5
                width: 64 bits
                clock: 33MHz
                capabilities: bus_master cap_list fb
                configuration: depth=32 driver=amdgpu latency=0 resolution=1920,1200
                resources: iomemory:fc0-fbf iomemory:fc0-fbf irq:48 memory:fce0000000-fcefffffff memory:fcf0000000-fcf01fffff ioport:1000(size=256) memory:d0300000-d037ffff
```

**Comment by js6pak** [comment]
Looks like #343806? Do you have this issue in any other tauri/electron/chromium apps?
Current commit works fine for me on my rdna3 amd gpu

**Comment by KiaraGrouwstra** [comment]
that seems my issue yes. i could retry once 24.11 hits and i've caught up on mesa versions, tho hopefully someone else can test before.

**Comment by KiaraGrouwstra** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 348387`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>restic-browser</li>
  </ul>
</details>

**Comment by KiaraGrouwstra** [comment]
thanks! it looks like it's working well for me now :)

**Review by KiaraGrouwstra — APPROVED** [review]

File: pkgs/by-name/re/restic-browser/package.nix [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    rev = "v${version}";
+    hash = "sha256-KE9pa4P6WyzNo3CxPKgREb6EEkUEQSuhihn938XN45A=";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "tauri-plugin-window-state-0.1.1" = "sha256-Mf2/cnKotd751ZcSHfiSLNe2nxBfo4dMBdoCwQhe7yI=";
+    };
+  };
```
Comment (SuperSandro2000) [inline pkgs/by-name/re/restic-browser/package.nix:33]: This should be using cargoVendor to avoid committing the big lock files.

**Pull request merged**